### PR TITLE
[Release-1.7] Update Proxy SHA (#26030)

### PR DESCRIPTION
**Pull request by gargnupur**
Signed-off-by: gargnupur <gargnupur@google.com>

Update Proxy SHA after the fix in proxy

Signed-off-by: gargnupur <gargnupur@google.com>

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

**Comment by jwendell** [comment]
@gargnupur can you fix the conflicts?

File: istio.deps [hunk 1]
```diff
@@ -4,6 +4,6 @@
     "name": "PROXY_REPO_SHA",
     "repoName": "proxy",
     "file": "",
-    "lastStableSHA": "8baa35b2a195134f697a7d821dd38c5aab2188e8"
+    "lastStableSHA": "8507cf040986885e45585a59a71f52dd3c6d9743"
```
Comment (jwendell) [inline istio.deps:7]: This is outdated, update please

**Comment by gargnupur** [comment]
@jwendell : looks like failure in lint is coming from https://github.com/istio/proxy/commit/422ba7d52af4e9e12ae75497f0d7483a565908b1... do you know anything about this automated updated?

**Comment by jwendell** [comment]
That will be fixed by https://github.com/istio/istio/pull/26278
When it merges, we can `/retest` this one.

**Pull request merged**